### ISSUE TITLE
[SUPPORT] Composing dynamic styles + Themes + Typescript

### ISSUE THREAD

**Issue by jacek213** [emotion-js/emotion#1670]
I'm trying to share styles between two components. Those styles use values from my theme. I've just tried something like:

```javascript
import styled, {ThemeProps} from "styled"
import {Link} from "react-router-dom"
import {css} from '@emotion/core'

const commonStyles = (props: ThemeProps) => css`
  cursor: pointer;
  text-align: center;
  font-size: ${props.theme.fontSizes[16]};
  font-weight: ${props.theme.fontWeights.bold};
  color: ${props.theme.colors.white};
  background-color: ${props.theme.colors.blue};
`

export const CtaButton = styled.button`
  ${commonStyles}
`
export const CtaLink = styled(Link)`
  ${commonStyles}
`

```

... but now, when attempting to use the component, I'm getting error because I haven't passed the `theme` prop:

`<CtaLink to={LOGIN_PATH}>sign in</CtaLink>` => `Property 'theme' is missing in type '{ children: string; to: string; }' but required in type 'ThemeProps'.`

My styled function is typed in a way suggested in the docs:

```javascript
import styled, {CreateStyled} from "@emotion/styled"
import {theme} from "./theme"

type Theme = typeof theme

interface ThemeProps {
  theme: Theme
}

export {theme, Theme, ThemeProps}

export default styled as CreateStyled<Theme>


```

**Comment by Andarist** [comment]
Sorry for the inconvenience but I'm afraid that I won't be able to help you with this one right now. v10 typings are rather complex and have some quirks - they have been completely reworked and the rewrite is available in v11 (still in beta). We plan to release v11 this month (no promises though!) and I believe that migrating to it will resolve your issue. I also encourage you to try out currently published v11 version - it's very much usable, doesn't have many breaking changes and we don't plan to make any significant breaking changes either. You could only maybe hold off until we release this https://github.com/emotion-js/emotion/pull/1609 as it is concerned with TS and Theme so affects you quite a bit.

**Comment by jacek213** [comment]
Thank you for a detailed answer @Andarist :) I've subscribed to #1609 and will test it out as soon as it's merged.

**Comment by Andarist** [comment]
Should be available in a moment in 11.0.0-next.8

Please give it a spin and if you find any issues just report them so we can fix them before the final release.
